infra/modules/service/step-functions.tf

```diff
@@ -0,0 +1,110 @@
+# docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_log_group
+resource "aws_cloudwatch_log_group" "sfn_cron_job" {
+  for_each = {
+    for index, sfn_vars in var.sfn_vars :
+    sfn_vars.name => sfn_vars
+  }
+
+  name_prefix = "/aws/vendedlogs/states/${var.service_name}-${each.value.name}"
+
+  # Conservatively retain logs for 5 years.
+  # Looser requirements may allow shorter retention periods
+  retention_in_days = 1827
+
+  # TODO(https://github.com/navapbc/template-infra/issues/164) Encrypt with customer managed KMS key
+  # checkov:skip=CKV_AWS_158:Encrypt service logs with customer key in future work
+}
+
+# docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sfn_state_machine
+resource "aws_sfn_state_machine" "sfn_cron_job" {
+  for_each = {
+    for index, sfn_vars in var.sfn_vars :
+    sfn_vars.name => sfn_vars
+  }
+
+  name     = "${var.service_name}-${each.value.name}"
+  role_arn = module.service.task_role_arn
+
+  definition = jsonencode({
+    "StartAt" : "ExecuteECSTask",
+    "States" : {
+      "ExecuteECSTask" : {
+        "Type" : "Task",
+        # docs: https://docs.aws.amazon.com/step-functions/latest/dg/connect-ecs.html
+        "Resource" : "arn:aws:states:::ecs:runTask.sync",
+        "Parameters" : {
+          "Cluster" : module.service.cluster_arn,
+          "TaskDefinition" : module.service.task_definition_arn,
```

Comment:
I don't understand how this works. `module.service` isn't defined anywhere in this module.

Reply:
Yeah thats WIP still, its a straight copy from grants.gov where the file is in a different place. Simple change to correct it, I just havent yet ^^